### ISSUE TITLE
separate convert with stringsAsFactors = FALSE

### ISSUE THREAD

**Issue by kismsu** [hadley/tidyr#99]
Hi, I know that you are not a big fan of stringsAsFactors. However 

```r
df <- data.frame(x = c("a.b", "a.d", "b.c"))
df %>% 
    separate(x, c("A", "B"), convert = TRUE) %>% 
    str()
'data.frame':	3 obs. of  2 variables:
 $ A: Factor w/ 2 levels "a","b": 1 1 2
 $ B: Factor w/ 3 levels "b","c","d": 1 3 2
```
Would it be a good idea to send stringsAsFactors = FALSE to `separate`?

**Comment by hadley** [comment]
Hmmm, I had `lapply(l, type.convert, as.is = FALSE)` and I don't understand why. I'm almost certain it should be `TRUE`.

**Comment by hadley** [comment]
Fixed in c77b42e377ede124d1e1c96295883e14fd424391
